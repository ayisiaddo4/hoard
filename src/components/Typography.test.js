/* eslint-env jest */

import React from 'react';
import T from 'components/Typography';

import renderer from 'react-test-renderer';
describe('<Typography/>', () => {
  test('<T.Light/>', () => {
    const tree = renderer.create(<T.Light>Light</T.Light>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.Heading/>', () => {
    const tree = renderer.create(<T.Heading>Heading</T.Heading>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.SubHeading/>', () => {
    const tree = renderer
      .create(<T.SubHeading>SubHeading</T.SubHeading>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.GrayedOut/>', () => {
    const tree = renderer.create(<T.GrayedOut>GrayedOut</T.GrayedOut>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.SemiBold/>', () => {
    const tree = renderer.create(<T.SemiBold>SemiBold</T.SemiBold>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.Small/>', () => {
    const tree = renderer.create(<T.Small>Small</T.Small>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.ButtonText/>', () => {
    const tree = renderer
      .create(<T.ButtonText>ButtonText</T.ButtonText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.Price/>', () => {
    const tree = renderer.create(<T.Price>Price</T.Price>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.PriceLarge/>', () => {
    const tree = renderer
      .create(<T.PriceLarge>PriceLarge</T.PriceLarge>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.PriceHeading/>', () => {
    const tree = renderer
      .create(<T.PriceHeading>PriceHeading</T.PriceHeading>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.TitleAlternate/>', () => {
    const tree = renderer
      .create(<T.TitleAlternate>TitleAlternate</T.TitleAlternate>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.SemiBoldAlternate/>', () => {
    const tree = renderer
      .create(<T.SemiBoldAlternate>SemiBoldAlternate</T.SemiBoldAlternate>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.SubtitleAlternate/>', () => {
    const tree = renderer
      .create(<T.SubtitleAlternate>SubtitleAlternate</T.SubtitleAlternate>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.SmallAlternate/>', () => {
    const tree = renderer
      .create(<T.SmallAlternate>SmallAlternate</T.SmallAlternate>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<T.Paragraph/>', () => {
    const tree = renderer.create(<T.Paragraph>Paragraph</T.Paragraph>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
