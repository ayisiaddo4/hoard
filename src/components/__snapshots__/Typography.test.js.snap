// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography/> <T.ButtonText/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 12,
        "letterSpacing": 0,
        "lineHeight": 18,
      },
      undefined,
    ]
  }
>
  ButtonText
</Text>
`;

exports[`<Typography/> <T.GrayedOut/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#5f627d",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  GrayedOut
</Text>
`;

exports[`<Typography/> <T.Heading/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 28,
        "fontWeight": "bold",
        "letterSpacing": -1,
      },
      undefined,
    ]
  }
>
  Heading
</Text>
`;

exports[`<Typography/> <T.Light/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  Light
</Text>
`;

exports[`<Typography/> <T.Paragraph/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  Paragraph
</Text>
`;

exports[`<Typography/> <T.Price/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 28,
        "letterSpacing": 1,
      },
      undefined,
    ]
  }
>
  Price
</Text>
`;

exports[`<Typography/> <T.PriceHeading/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontFamily": "HelveticaNeue",
        "fontSize": 44,
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  PriceHeading
</Text>
`;

exports[`<Typography/> <T.PriceLarge/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontFamily": "HelveticaNeue",
        "fontSize": 32,
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  PriceLarge
</Text>
`;

exports[`<Typography/> <T.SemiBold/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  SemiBold
</Text>
`;

exports[`<Typography/> <T.SemiBoldAlternate/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  SemiBoldAlternate
</Text>
`;

exports[`<Typography/> <T.Small/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 12,
        "letterSpacing": 0,
        "lineHeight": 18,
      },
      undefined,
    ]
  }
>
  Small
</Text>
`;

exports[`<Typography/> <T.SmallAlternate/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 12,
        "letterSpacing": 1,
      },
      undefined,
    ]
  }
>
  SmallAlternate
</Text>
`;

exports[`<Typography/> <T.SubHeading/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 18,
        "letterSpacing": 0,
      },
      undefined,
    ]
  }
>
  SubHeading
</Text>
`;

exports[`<Typography/> <T.SubtitleAlternate/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#00EC9E",
        "fontFamily": "HelveticaNeue",
        "fontSize": 12,
        "letterSpacing": 1,
      },
      undefined,
    ]
  }
>
  SubtitleAlternate
</Text>
`;

exports[`<Typography/> <T.TitleAlternate/> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#17181e",
        "fontFamily": "HelveticaNeue",
        "fontSize": 28,
        "letterSpacing": 0,
        "lineHeight": 32,
      },
      undefined,
    ]
  }
>
  TitleAlternate
</Text>
`;
